src/pages/costModels/costModel/dialog.test.tsx

```diff
@@ -1,45 +1,40 @@
-import { Alert } from '@patternfly/react-core';
-import { shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
 import React from 'react';
 
 import Dialog from './dialog';
 
 const defaultProps = {
   onClose: jest.fn(),
   onProceed: jest.fn(),
+  isOpen: true,
   title: 'Test dialog',
   body: <div>This is a test dialog body</div>,
   t: (text: string) => text,
 };
 
-test('dialog has exclamation triangle icon in the title', () => {
-  const view = shallow(<Dialog {...defaultProps} />);
-  const icon = view.prop('header').props.children[0];
-  expect(icon.type.displayName).toBe('ExclamationTriangleIcon');
-  expect(icon.props).toEqual({
-    color: 'orange',
-    noVerticalAlign: false,
-    size: 'sm',
-  });
```

Comment:
Maybe I'm missing something, but how does the snapshot confirm there is a triangle in the dom? There are only certain conditions where that is displayed.

Not sure the previous test necessarily needed to open the dialog, -- we can test other parts of the code.
Previously, we were testing for the icon's displayName, but the snapshot doesn't give me confidence if the triangle was actually displayed or not.

Reply:
If you look at the snapshot, it contains an svg with the color orange and a path with a d attribute. Those values are what are actually displaying the triangle. The previous tests didn't actually test it was there.. just that a property asking for it was sent to the component. This test is really testing the DOM so what the user sees, if you plug the DOM from the snap into codepen, you'll see what I mean: https://codepen.io/cpdallas/pen/oNELReo

I don't think it matters, but from what I can see, the triangle would always be there when using dialog.tsx:

```
  return (
    <Modal
      aria-label={title}
      header={
        <Title headingLevel="h1" size={TitleSizes['2xl']}>
          <ExclamationTriangleIcon color="orange" /> {title}
        </Title>
      }
```

Reply:
Perhaps we don't need to test for the triangle, since it does not appear to be dynamic in the code. I was thinking of a dynamic triangle icon shown with certain sources.

Probably would be better to test if an alert is displayed? The error alerts are dynamic.

Reply:
Lines 31-35 test for error alert

Reply:
Ok, I see that